### PR TITLE
feat: This is the base branch for commute details related changes

### PR DESCRIPTION
**Pull request by suyashpatil78**
### Description
copilot:summary

copilot:poem

### Walkthrough
copilot:walkthrough

## Clickup
app.clickup.com

## Code Coverage
Please add code coverage here

## UI Preview
Please add screenshots for UI changes

### CODE REVIEW

**Comment by suyashpatil78** [comment]
Would be fixing unit tests separately in subsequent PRs

File: src/global.scss [hunk 1]
```diff
@@ -1002,6 +1002,15 @@ ion-modal.payment-mode-modal {
   }
 }
 
+ion-modal.add-location-modal {
+  &::part(content) {
+    border-radius: 16px 16px 0 0 !important;
```
Comment (howdysuraj) [inline src/global.scss:1007]: can we get rid of this !important and try to use the css specificity  for targeting the element if possible

File: src/app/shared/components/route-selector/route-selector.component.ts [hunk 2]
```diff
@@ -85,10 +99,10 @@ export class RouteSelectorComponent implements OnInit, ControlValueAccessor, OnD
     this.onChangeSub.unsubscribe();
   }
 
-  customDistanceValidator(control: AbstractControl) {
+  customDistanceValidator(control: AbstractControl): { invalidDistance: boolean } {
     const passedInDistance = control.value && +control.value;
```
Comment (howdysuraj) [inline src/app/shared/components/route-selector/route-selector.component.ts:103]: @suyashpatil78 are we trying to control.value to number using +, can we use some simple and much readable approach for number conversion , it seems everyone is not aware of this shorthand.

**Review by howdysuraj — APPROVED** [review]

File: src/global.scss (hunk 1 above)
Comment (suyashpatil78) [reply src/global.scss:1007]: actually we don't need !important removing this

File: src/app/shared/components/route-selector/route-selector.component.ts (hunk 2 above)
Comment (suyashpatil78) [reply src/app/shared/components/route-selector/route-selector.component.ts:103]: sure Suraj, we can use `parseFloat`

**Pull request merged**